rwengine/include/engine/GameInputState.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef RWENGINE_GAMEINPUTSTATE_HPP
+#define RWENGINE_GAMEINPUTSTATE_HPP
+
+struct GameInputState
+{
+	/// Inputs that can be controlled
+	/// @todo find any sensible values
+	enum Control {
+		/* On Foot */
+		FireWeapon = 0,
+		NextWeapon,
+		NextTarget = NextWeapon,
+		LastWeapon,
+		LastTarget = LastWeapon,
+		GoForward,
+		GoBackwards,
+		GoLeft,
+		GoRight,
+		ZoomIn,
+		ZoomOut,
+		EnterExitVehicle,
+		ChangeCamera,
+		Jump,
+		Sprint,
+		AimWeapon,
+		Crouch,
+		Walk,
+		LookBehind,
+
+		/* In Vehicle */
+		VehicleFireWeapon = FireWeapon,
+		VehicleAccelerate,
```

Comment:
On IRC we came to the conclusion that mapping the logical state to the correct button state for the script was the way to go.